ocpp_codec/encoders.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Polyconseil SAS. All rights reserved.
+"""Collection of classes that can be used to encode and decode OCPP data.
+
+Encoders are expected to convert the incoming value to another type (e.g.: from a string to a 'datetime.datetime').
+They should raise an appropriate instance of a 'BaseOCPPError' if the input value does not meet the appropriate
+criteria.
+
+Encoders shouldn't run any type checking when parsing ('from_json'), but should when serializing ('to_json'). This is
+because there might be several Python types that can be coerced into the correct OCPP-JSON type required. Encoders are
+responsible for raising a 'TypeConstraintViolationError' error if they cannot process the input.
+
+Encoders are assigned to a dataclass field through its metadata 'encoder' key. Only a single encoder is accepted.
+"""
+import datetime
+import typing
+
+import dateutil.parser
+import pytz
+
+from ocpp_codec import errors
+
+
+class BaseEncoder:
```

Comment:
:thinking: It may be one of the good reason to use a proper `abc.ABCMeta` and `@abstractmethod`.

Reply:
I've tried something, let me know what you think about it. It doesn't change much but it does make the intent clearer.